latex/programming_interface.tex

```diff
@@ -179,7 +179,8 @@ \subsection{Common by-value semantics}
 
 Some \gls{sycl-runtime} classes will have additional behavior associated with copy, movement, assignment or destruction semantics. If these are specified they are in addition to those specified above unless stated otherwise.
 
-Each of the runtime classes mentioned above must provide a common interface of special member functions and member functions in order to fulfil the copy, move, destruction and equality requirements.
+Each of the runtime classes mentioned above must provide a common interface of special member functions and member functions in order to fulfil the copy, move, destruction and equality requirements.  If all of the special member functions are inlined, public and defaulted, none of them should be explicitly declared (Rule of Zero); otherwise, all of them should be explicitly declared (Rule of Five).
```

Comment:
I think we need a link to some reference for rule of Zero and rule of Five, not all SYCL readers are necessarily well versed in C++

Reply:
I agree we should define what these terms mean and the justification for them to avoid confusion. It is defined in the C++ core guidelines, but I don't believe they have a fixed revision we can reference, the ISO C++ standard is also hard to reference on this rule as the wording is scattered across a number of sections.

Perhaps we should have a glossary term that describes the rule of zero and the rule of five in more detail, then anywhere we mention it we can just use the glossary term?

Reply:
I'm moving the definitions of "rule of five" and "rule of zero" to the glossary.

Reply:
I don't think we should provide a link to the core guidelines, as this spec is for SYCL implementers.  How users write their code, other than meeting requirements, is beyond what this specification covers.